internal/v2/cert_verifier/cert_verifier.go

```diff
@@ -0,0 +1,62 @@
+// Certificate Verification that offloads operations to S2Av2.
+package certverifier
+
+import (
+	"crypto/x509"
+	"fmt"
+	"time"
+)
+
+// VerifyClientCertificateChain returns any errors found during certificate verification.
+func VerifyClientCertificateChain(local_id string, pool *x509.CertPool) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
```

Comment:
nit: in this implementation, we're comparing `local_id` to the common name in the peer leaf cert, so should we rename the variable to `expectedCommonName`?

Similarly for the server cert API.

Reply:
Agreed, thanks!